src/lib.rs

```diff
@@ -63,7 +63,9 @@ const ENV_CERT_FILE: &str = "SSL_CERT_FILE";
 /// Returns None if SSL_CERT_FILE is not defined in the current environment.
 ///
 /// If it is defined, it is always used, so it must be a path to a real
-/// file from which certificates can be loaded successfully.
+/// file from which certificates can be loaded successfully. However,
```

Comment:
I'd skip the "However" here, which doesn't seem entirely appropriate? Maybe rephrase as:

> While parsing, [`rustls_pemfile::certs()`] parser will ignore parts of the file which are not part of a certificate.

Reply:
"However" I chose because I'm trying to say certificate loading needs to be successful but the fact that parts of the file may be ignored (e.g. because the wrong certificate format is used or a certificate is corrupted) can mean that we report success/`Ok(_)` when really the certificate failed to parse, at least from a user's perspective.

What about this:

> While parsing, [rustls_pemfile::certs()] parser will ignore parts of the file which are not considered part of a certificate. Certificates which are not in the right format (PEM) or are otherwise corrupted may get ignored silently.

I also wonder if this should we document this in the public part of the documentation too. Perhaps we could clarify what format is expected to make it less likely people will use the wrong format. Something like:

Be sure certificates are in PEM format:

```
<insert example cert in PEM format>
```

Reply:
What's an error and what's ignored seems a bit arbitrary to me.

This is ignored

```
----BEGIN CERTIFICATE-----
^ missing '-' at start
<content>
-----END CERTIFICATE-----
```

This is an error:

```
-----BEGIN CERTIFICATE-----
<content>
-----END CERTIFICATE----
                        ^ missing '-'
```

This is ignored:

```
-----BEGIN CERTIFICATE-----
<content>
-----END CERTIFICATE----
                        ^ missing '-' AND '\n'
```

This is ignore:

```
-----BEGIN CERTIFICATE-----
<content>
-----END CERTIFICATE-----
                         ^ missing '\n'
```

Reply:
I think we'd probably take PRs to make all of those errors, but the main goal in that crate was to properly ignore things that don't look like PEM objects at all which has been sighted in the wild. I guess these cases really haven't shown up.

All of your documentation suggestions sound good to me!

Reply:
(Still pending)

Reply:
PEM appears to use the encoding defined in [RFC7468](https://datatracker.ietf.org/doc/html/rfc7468). Ignoring invalid sections of the file is part of the encoding but, if I read the grammar right, the line feed at the end of `-----END CERTIFICATE-----` is explicitly optional. So, it should probably be made optional in RusTLS parser too.

Reply:
Thanks for looking into that! Would be good to file a followup issue and/or PR for that.

Reply:
Added it to my todo list.